src/cmd/services/m3coordinator/ingest/m3msg/ingest.go

```diff
@@ -200,8 +209,27 @@ func (op *ingestOp) resetWriteQuery() error {
 func (op *ingestOp) resetTags() error {
 	op.it.Reset(op.id)
 	op.q.Tags.Tags = op.q.Tags.Tags[:0]
+	op.q.Tags.Opts = op.tagOpts
 	for op.it.Next() {
 		name, value := op.it.Current()
+
+		// TODO_FIX_GRAPHITE_TAGGING: Using this string constant to track
+		// all places worth fixing this hack. There is at least one
+		// other path where flows back to the coordinator from the aggregator
+		// and this tag is interpreted, eventually need to handle more cleanly.
+		if bytes.Equal(name, downsample.MetricsOptionIDSchemeTagName) {
+			if bytes.Equal(value, downsample.GraphiteIDSchemeTagValue) &&
+				op.q.Tags.Opts.IDSchemeType() != models.TypeGraphite {
```

Comment:
Oof yuck :p
small nit: probably be slightly more performant to check for scheme type first? Or to get that as a bool earlier to avoid function calls?